### ISSUE TITLE
Crash due to `IGNORED_SUBDIRECTORIES` introduced in PR #23999

### ISSUE THREAD

**Issue by jpsim** [bazelbuild/bazel#24611]
### Description of the bug:

I'm experiencing a new crash in 8.0.0rc8 that doesn't occur in 7.4.1 or 8.0.0rc1, seemingly introduced in https://github.com/bazelbuild/bazel/pull/23999.

Here's what the crash stack trace looks like:

```console
$ USE_BAZEL_VERSION=8.0.0rc8 bazel run cli
Starting local Bazel server and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
Analyzing: target //:cli (72 packages loaded, 1277 targets configured)
    Fetching repository @@apple_support++apple_cc_configure_extension+local_config_apple_cc; starting
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'IGNORED_SUBDIRECTORIES:@@rules_swift_package_manager++swift_deps+swiftpkg_asc_swift' (requested by nodes 'PACKAGE_LOOKUP:@@rules_swift_package_manager++swift_deps+swiftpkg_asc_swift//')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:547)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:435)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:131)
	at com.google.devtools.build.lib.cmdline.IgnoredSubdirectories.of(IgnoredSubdirectories.java:103)
	at com.google.devtools.build.lib.skyframe.IgnoredSubdirectoriesValue.of(IgnoredSubdirectoriesValue.java:45)
	at com.google.devtools.build.lib.skyframe.IgnoredSubdirectoriesFunction.compute(IgnoredSubdirectoriesFunction.java:191)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:467)
	... 7 more
```

Initially reported here: https://github.com/cgrindel/rules_swift_package_manager/issues/1394

### Which category does this issue belong to?

Core, Loading API

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

To reproduce (assuming bazelisk & Xcode are installed):

**`MODULE.bazel`**

```python
bazel_dep(name = "apple_support", version = "1.17.1")
bazel_dep(name = "rules_swift", version = "2.2.4")
bazel_dep(name = "rules_swift_package_manager", version = "0.42.0")

swift_deps = use_extension(
    "@rules_swift_package_manager//:extensions.bzl",
    "swift_deps",
)
swift_deps.from_package(
    resolved = "//:Package.resolved",
    swift = "//:Package.swift",
)
use_repo(
    swift_deps,
    "swift_package",
    "swiftpkg_asc_swift",
)
```

**`Package.swift`**

```swift
// swift-tools-version: 5.10

import PackageDescription

let package = Package(
    name: "basic_bazel_swift_cli",
    dependencies: [
        .package(url: "https://github.com/aaronsky/asc-swift", from: "1.2.0"),
    ]
)
```

**`main.swift`**

```swift
import AppStoreConnect

print("Hello world")
```

**`BUILD`**

```python
load("@rules_swift//swift:swift_binary.bzl", "swift_binary")

swift_binary(
    name = "cli",
    srcs = ["main.swift"],
    deps = [
        "@swiftpkg_asc_swift//:AppStoreConnect",
    ],
)
```

Run the following:

```console
$ swift package resolve
$ USE_BAZEL_VERSION=7.4.1 bazel run cli
INFO: Analyzed target //:cli (112 packages loaded, 4071 targets configured).
INFO: Found 1 target...
Target //:cli up-to-date:
  bazel-bin/cli
INFO: Elapsed time: 195.762s, Critical Path: 189.37s
INFO: 55 processes: 27 internal, 23 darwin-sandbox, 5 worker.
INFO: Build completed successfully, 55 total actions
INFO: Running command line: bazel-bin/cli
Hello world
$ USE_BAZEL_VERSION=8.0.0rc8 bazel run cli
Starting local Bazel server and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
Analyzing: target //:cli (72 packages loaded, 1277 targets configured)
    Fetching repository @@apple_support++apple_cc_configure_extension+local_config_apple_cc; starting
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'IGNORED_SUBDIRECTORIES:@@rules_swift_package_manager++swift_deps+swiftpkg_asc_swift' (requested by nodes 'PACKAGE_LOOKUP:@@rules_swift_package_manager++swift_deps+swiftpkg_asc_swift//')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:547)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:435)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:131)
	at com.google.devtools.build.lib.cmdline.IgnoredSubdirectories.of(IgnoredSubdirectories.java:103)
	at com.google.devtools.build.lib.skyframe.IgnoredSubdirectoriesValue.of(IgnoredSubdirectoriesValue.java:45)
	at com.google.devtools.build.lib.skyframe.IgnoredSubdirectoriesFunction.compute(IgnoredSubdirectoriesFunction.java:191)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:467)
	... 7 more
```

### Which operating system are you running Bazel on?

macOS 15.1.1 (24B91)

### What is the output of `bazel info release`?

release 8.0.0rc8

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.

8bfb921004ade0a05e80dfbbaf0083512574a424

### Any other information, logs, or outputs that you want to share?

Initially reported here: https://github.com/cgrindel/rules_swift_package_manager/issues/1394

**Comment by brentleyjones** [comment]
@bazel-io flag

**Comment by meteorcloudy** [comment]
Probably related to https://github.com/aaronsky/asc-swift/blob/main/.bazelignore

**Comment by meteorcloudy** [comment]
We are going to release Bazel 8, this will be fixed in a patch release coming shortly

**Comment by lberki** [comment]
I don't have access to a Mac to reproduce with (hence, no Xcode) and the above reproduction does seem to need it. Is this reproducible somehow without Xcode?

I'm fairly certain @meteorcloudy is right on the money but I only have very limited time this evening and the trivial attempt of using that `.bazelignore` file, putting a glob in a BUILD file then running `bazel query` doesn't seem to tickle it with either 8.0.0rc8 or HEAD.

**Comment by jpsim** [comment]
Thank you very much for the quick fix and workaround!

**Comment by iancha1992** [comment]
@bazel-io fork 8.1.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.1.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/8.1.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.1.0rc1. Thanks!
